frontend/mulighetsrommet-veileder-flate/src/components/joyride/DetaljerJoyride.tsx

```diff
@@ -0,0 +1,150 @@
+
+interface Props {
+  setDelMedBrukerModal: (state: boolean) => void;
+  opprettAvtale: boolean;
+}
+export function DetaljerJoyride({ setDelMedBrukerModal, opprettAvtale }: Props) {
+  const [state, setState] = useState({
+    run: false,
+    loading: false,
+    stepIndex: 0,
+  });
+
+  useEffect(() => {
+    //viser joyride ved første load
+    return window.localStorage.getItem('joyrideDetaljer') === null
+      ? window.localStorage.setItem('joyrideDetaljer', 'true')
+      : window.localStorage.setItem('joyrideDetaljer', 'false');
+  }, []);
+
+  useEffect(() => {
+    if (window.localStorage.getItem('harVistJoyrideOpprettAvtale') === null && opprettAvtale) {
+      return window.localStorage.setItem('harVistJoyrideOpprettAvtale', 'true');
+    } else if (!opprettAvtale && window.localStorage.getItem('harVistJoyrideOpprettAvtale') === null) {
+      return window.localStorage.setItem('harVistJoyrideOpprettAvtale', 'false');
+    }
+  }, []);
+
+  const steps: Step[] = [
+    {
+      title: 'Tiltaksgjennomføring',
+      content: 'Her kan du lese mer om tiltaket.',
+      placement: 'auto',
+      target: '#tiltaksgjennomforing_detaljer',
+      disableBeacon: true,
+    },
+    {
+      content: 'Du kan bytte fane her og lese mer informasjon om tiltaket.',
+      placement: 'auto',
+      target: '#fane_liste',
+      disableBeacon: true,
+    },
+    {
+      content: 'I dette området kan du lese mer nøkkelinfromasjon.',
+      placement: 'auto',
+      target: '#sidemeny',
+      disableBeacon: true,
+    },
+    {
+      content: 'For tiltak med {MER TEKST HER} kan du opprette avtale.',
+      placement: 'auto',
+      target: '#opprett-avtale-knapp',
+      disableBeacon: true,
+    },
+    {
+      content: 'Du kan også dele tiltaket med bruker via dialogen! Klikk neste for å se.',
+      placement: 'auto',
+      target: '#deleknapp',
+      disableBeacon: true,
+    },
+    {
+      content: 'Dette er teksten brukeren vil se i dialogen etter at tiltaket er delt.',
+      placement: 'auto',
+      styles: {
+        options: {
+          zIndex: 10000,
+        },
+      },
+      target: '#deletekst',
+      disableBeacon: true,
+    },
+    {
+      content: 'Du kan også legge til mer tekst ved å klikke her.',
+      placement: 'auto',
+      styles: {
+        options: {
+          zIndex: 10000,
+        },
+      },
+      target: '#personlig_melding_btn',
+      disableBeacon: true,
+    },
+  ];
+
+  const handleJoyrideCallback = (data: any) => {
+    const { action, index, status, type } = data;
+    const nextStepIndex = index + (action === ACTIONS.PREV ? -1 : 1);
+
+    if (!opprettAvtale) {
+      //hvis det ikke er tiltak med opprett avtale, settes opprett avtale-steps til ikke vist i localStorage
+      window.localStorage.setItem('harVistJoyrideOpprettAvtale', 'false');
+      //hopper over steget med opprett avtale for at den skal kjøre videre
+      if (index === 3 && [ACTIONS.NEXT]) {
```

Comment:
Disse indeksene ser ganske skjøre ut mtp. eventuelle endringer, men er ikke sikkert vi trenger å tenke ut noen bedre løsning hvis vi tenker at joyrides bare er en midlertidig greie.

Hvis du er interessert så kan vi evt. ta en session sammen for å se om vi klarer å pønske ut et slags api basert på id'er i stedet (som mapper tilbake til indeksene under panseret). Kommer vel igjen an på om vi anser joyrides som en kortlevende feature, eller om vi ønsker å vedlikeholde disse like lenge som applikasjonen selv.

Reply:
Gjerne det!